### DISCUSSION TITLE
pyenv install 3.11-dev gets stuck on macos m1

### DISCUSSION THREAD

**Discussion by ssbarnea** [pyenv/pyenv#2408] [Q&A] [answered]
I tried to upgrade my py3.11 from b3 to newer but it seems that pyenv gets stuck without giving any feedback. I also checked the activity monitor and I do not think there is a process doing anything.
```
$ pyenv install 3.11-dev
pyenv: /Users/ssbarnea/.pyenv/versions/3.11-dev already exists
continue with installation? (y/N) y
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Cloning https://github.com/python/cpython...
Installing Python-3.11-dev...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk
```

I think that pyenv should mention the log file on console as it took me a lot of time to spot the bash process that it was using ang grab the log file from there.

```
$ tail -f /private/var/folders/32/1xrphgzd4xv777syxjtkpdw80000gn/T/python-build.20220706113943.35209.log
creating build/scripts-3.11
copying and adjusting /private/var/folders/32/1xrphgzd4xv777syxjtkpdw80000gn/T/python-build.20220706113941.35163/Python-3.11-dev/Tools/scripts/pydoc3 -> build/scripts-3.11
copying and adjusting /private/var/folders/32/1xrphgzd4xv777syxjtkpdw80000gn/T/python-build.20220706113941.35163/Python-3.11-dev/Tools/scripts/idle3 -> build/scripts-3.11
copying and adjusting /private/var/folders/32/1xrphgzd4xv777syxjtkpdw80000gn/T/python-build.20220706113941.35163/Python-3.11-dev/Tools/scripts/2to3 -> build/scripts-3.11
changing mode of build/scripts-3.11/pydoc3 from 644 to 755
changing mode of build/scripts-3.11/idle3 from 644 to 755
changing mode of build/scripts-3.11/2to3 from 644 to 755
renaming build/scripts-3.11/pydoc3 to build/scripts-3.11/pydoc3.11
renaming build/scripts-3.11/idle3 to build/scripts-3.11/idle3.11
renaming build/scripts-3.11/2to3 to build/scripts-3.11/2to3-3.11
```

Shortly nothing really happening.

I mentioned that I also installed pyenv from head, just to be sure it was not a bug that was already fixed, but still the same. Clearly it was able to build once, when i installed it first time.

**Comment by native-api** [comment, marked as answer]
Please report this as an issue and provide the necessary diagnostic information as per the issue template.

**Comment by native-api** [comment]
> I think that pyenv should mention the log file on console

It does so if the build fails.

Actually, aren't you able to terminate a hung build with Ctrl-C?

**Reply by ssbarnea** [reply]
Ctrl-C worked but it did not provide more info than what I posted.

**Comment by hitblast** [comment]
Hello! I tried replicating your situation by installing `3.11-dev` on my M1 MacBook Air and it did successfully install.

<img width="1030" alt="Screenshot 2022-07-15 at 11 07 55 PM" src="https://user-images.githubusercontent.com/67194380/179273709-db8fbe7b-3477-4add-9e1a-ad0ab0332012.png">

Terminal: iTerm
macOS Version: Monterey 12.4
Command Line Tools for Xcode Version: 13.4

The process was relatively fast, considering it took less than a minute to clone CPython (the Internet matters in this case) and where I did see some long processing periods is when `conftest` was running (the top bar, which says `zsh` in the screenshot, stayed at `conftest` for a really long time). One thing that is for sure is, you might have outdated dependencies. Consider updating them through homebrew and re-running the command.

**Reply by ssbarnea** [reply]
Brew is fully updated/upgraded and the problem can be reproduced with python-build locally. It takes about 2 minutes to build and building 3.11-b3 and b4 works without getting stuck, and even dev worked at least once, the first time. I wonder if there is something weird about the clone. Also it seems that the stuck situation is near the end, where it does some renames. 

I do have a tab open with the form to raise a bug, hopefully I can finish completing it this weekend.

I would have jumped to b4 but i hate to change python paths to all those vevns and vscode projects. Having the dev version one allows me to upgrade it without messing those.

**Reply by brondsem** [reply]
Did you figure anything out with this?  I am getting the same problem, on an intel mac.  I've created an issue at https://github.com/pyenv/pyenv/issues/2427 with trace details
